test/integrationTests/semanticTokensProvider.test.ts

```diff
@@ -60,19 +60,25 @@ suite(`SemanticTokensProvider: ${testAssetWorkspace.description}`, function () {
         }
 
         const activation = await activateCSharpExtension();
-        await testAssetWorkspace.restore();
 
         // Wait for workspace information to be returned
         let isWorkspaceLoaded = false;
 
-        const subscription = activation.eventStream.subscribe(event => {
-            if (event.type === EventType.WorkspaceInformationUpdated) {
-                isWorkspaceLoaded = true;
-                subscription.unsubscribe();
-            }
-        });
+        if (!activation.wasActive) {
+            const subscription = activation.eventStream.subscribe(event => {
```

Comment:
This won't stop execution. If you had a promise that resolved when `WorkspaceInformationUpdated` then you could await it to stop execution.

Reply:
that's a possibility as well, I was just trying to think about the minimal change to the test structure, but having a built in promise would work as well.

Reply:
Could still define the promise in the test.

```js
const workspaceLoadedPromise = new Promise(resolve =>
{
    const subscription = activation.eventStream.subscribe(event => {
        console.log(event);
        if (event.type === EventType.WorkspaceInformationUpdated) {
            subscription.unsubscribe();
            resolve(true);
        }
});
isWorkspaceLoaded = await workspaceLoadedPromise;
```

Reply:
Also, I now see the polling below that waits for isWorkspaceLoaded. please disregard my suggestion.